### PR TITLE
Sharing : Step five->move file_sharing_tab.js file from server repo to theme repo and cust…

### PR DESCRIPTION
**Pull request by TSI-ashishsharma**
move the file_sharing_tab.js file from the server repo to the theme repo and customize as per our requirement also add a store.js file for state management

### CODE REVIEW

**Comment by tsdicloud** [comment]
Hello Ashish,
unfortunately, we cannot keep the redesign like that. 
Remember that my guidance has been:
- extend vue.js/Jaavscript component (so that only the parts that we adapt are changed from a component
  and that we take over as much as possible of the upstream changes from Nextcloud automatically.
- If we "overwrite" a component, i.d. we keep the component with name, design and functions as it is and change only some lines, this stays a core change. if the .js filename, the namespace and the class name/ function name stay the same,
   the component/code has to stay at their original location.
- same is true for overwriting php code: if the php filename, the package name and the class name stay the same,
   the component has to stay at their original location.

To sum up, the more precise rule is:
- Extension: move to theming or app
- overwrite: stay in sync with the original code, i.e. don't move

Another important thinking is:
Theming means that we build a variation of the default Nextcloud appearance for our customer,
but not breaking the Nextcloud code.
Consequences:
1. If we implement a new view or a new component, it has to work with default Nextcloud style.
   Reason: if we want to bring the component upstream to Nextcoud standard, no Meganta-specific
   will be accepted
 2. all Magenta specific parts go into theming -> this is the purpose of a theme

**tl;dr:** 
- As long as you also copy a complete .js or and php file without some "extend(s)" statement in code
it does not make sense to shift it from their original place.
- Separate standard appearance from Magenta appearance with theme
 
Please tell also the team about the clarification and please care, we are enough in trouble with the customer

**Comment by tsdicloud** [comment]
Well known rule of thumbing is that copies of files that you want to overwrite MUST be in the äquivalent directory, this has not changed since 2021. So `magentacloud25/^custom^/apps` is an invalid try to avoid detection of impact for the overwrite, directors MUST be `magentacloud25/apps` to keep stable relationship to upstream code.

Invalid design, will unmark!
